spec/acceptance/dataset_items_spec.rb

```diff
@@ -0,0 +1,481 @@
+  end
+
+  delete '/datasets/:dataset_id/items/:id' do
+    dataset_id_param
+    dataset_item_id_param
+    let(:id) { dataset_item.id }
+    let(:dataset_id) { dataset.id }
+    let(:authentication_token) { no_access_token }
+    standard_request_options(:delete, 'DELETE (as no access user)', :forbidden, {expected_json_path: get_json_error_path(:permissions)})
+  end
+
+  delete '/datasets/:dataset_id/items/:id' do
+    dataset_id_param
+    dataset_item_id_param
+    let(:id) { dataset_item.id }
+    let(:dataset_id) { dataset.id }
+    let(:authentication_token) { invalid_token }
+    standard_request_options(:delete, 'DELETE (as invalid token)', :unauthorized, {expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+  delete '/datasets/:dataset_id/items/:id' do
+    dataset_id_param
+    dataset_item_id_param
+    let(:id) { dataset_item.id }
+    let(:dataset_id) { dataset.id }
+    standard_request_options(:delete, 'DELETE (as not logged in)', :unauthorized, {expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+  ################################
+  # FILTER
+  ################################
+
+  post '/dataset_items/filter' do
+    let(:authentication_token) { admin_token }
+    standard_request_options(:post, 'FILTER (as admin)', :ok,
+                             {response_body_content: ['200', '11'], expected_json_path: 'data/0/start_time_seconds', data_item_count: 1})
+  end
+
+  post '/dataset_items/filter' do
+    let(:authentication_token) { reader_token }
+    standard_request_options(:post, 'FILTER (as reader)', :ok,
+                             {response_body_content: ['200', '11'], expected_json_path: 'data/0/start_time_seconds', data_item_count: 1})
+  end
+
+  post '/dataset_items/filter' do
+    let(:authentication_token) { writer_token }
+    standard_request_options(:post, 'FILTER (as writer)', :ok,
+                             {response_body_content: ['200', '11'], expected_json_path: 'data/0/start_time_seconds', data_item_count: 1})
+  end
+
+  post '/dataset_items/filter' do
+    let(:authentication_token) { no_access_token }
+    standard_request_options(:post, 'FILTER (as no access)', :ok,
+                             {response_body_content: ['200'], expected_json_path: 'data', data_item_count: 0})
+  end
+
+  post '/dataset_items/filter' do
+    let(:authentication_token) { invalid_token }
+    standard_request_options(:post, 'FILTER (as invalid token)', :unauthorized, {expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+  # not logged in users can filter dataset items, but they won't get any items that they don't have permission for
+  post '/dataset_items/filter' do
+    standard_request_options(:post, 'FILTER (as not logged in)', :ok,
+                             {response_body_content: ['200'], expected_json_path: 'data', data_item_count: 0})
+  end
+
+  post '/dataset_items/filter' do
```

Comment:
this test would have more meaning if there were more than one item in the dataset - then it would actually be filtering out something

Reply:
Fixed, I think